forte/data/data_pack.py

```diff
@@ -1548,18 +1567,29 @@ def update(self, datapack: "DataPack"):
         #   better solution.
         self.__dict__.update(datapack.__dict__)
 
-    def _save_entry_to_data_store(self, entry: Entry):
+    def _save_entry_to_data_store(self, entry: Entry, attribute_data: Dict):
         r"""Save an existing entry object into DataStore"""
-        self._entry_converter.save_entry_object(entry=entry, pack=self)
+        self._entry_converter.save_entry_object(
+            entry=entry, pack=self, attribute_data=attribute_data
+        )
 
         if isinstance(entry, Payload):
-            if entry.modality == Modality.Text:
+            if (
+                getattr(Modality, attribute_data["modality_name"])
```

Comment:
That is because the `modality_name` is stored in the primitive form (`str`) in the datastore. That is why, when going through the conditions, we need to convert it back to its original form.

Reply:
there are two questions here:
1. let's try to use less hard code, such as `modality_name` here.
2. can we access the enum using the [name](https://docs.python.org/3/library/enum.html#programmatic-access-to-enumeration-members-and-their-attributes)? Using `getattr` looks hacky to me